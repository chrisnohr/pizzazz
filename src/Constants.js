import { makeBagOfLetters } from './scrabbleLogic/makeRandomLetters'

export const wordScoreDict = require('./word_score_dict.json');

export const wordScoreString = JSON.stringify(wordScoreDict);

export const letterDistribution = {
  'e':12, 'a':9, 'i':9, 'o':8, 'n':6, 'r':6, 't':6, 'l':4, 's':4, 'u':4, 'd':4, 'g':3, 'b':2,
  'c':2, 'm':2, 'p':2, 'f':2, 'h':2, 'v':2, 'w':2, 'y':2, 'k':1, 'j':1, 'x':1, 'q':1, 'z':1, '8':2
};

export const bagOfLetters = makeBagOfLetters(letterDistribution);

export const letterValues = {
  'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4, 'I': 1, 'J': 8, 'K': 5, 'L': 1, 'M': 3,
  'N': 1, 'O': 1, 'P': 3, 'Q': 10, 'R': 1, 'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8, 'Y': 4, 'Z': 10, '8': 0
}
