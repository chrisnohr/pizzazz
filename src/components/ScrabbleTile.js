import React, { Component } from 'react'
import styled from 'styled-components'

const LetterStyled = styled.div`
  border: 1px solid lightgray;
  border-radius: 0.5vw;
  color: #555;
  background-color: ${props => props.isValid ? '#e2fee2' : '#fffbf0'};
  font-weight: 200;
  font-size: 8vw;
  text-align: center;
  box-shadow: 1px 2px 1px 0 rgba(0, 0, 0, 0.05);

  @media screen and (min-width: 501px) {
      font-size: 40px;
      line-height: 65px;
      border-radius: 4px;
  }
`;

const NumberSub = styled.sub`
  line-height: normal;
  font-size: 2.6vw;
  font-weight: 400;
  vertical-align: -25%;

  @media screen and (min-width: 501px) {
    font-size:14.5px;
  }
`;

const ScrabbleTile = (props) => {
  console.log(props);
  const letter = props.letter;
  return (
    <LetterStyled isValid={ props.isValid }>
      { letter }
      <NumberSub>
        { props.letterValues[letter] }
      </NumberSub>
    </LetterStyled>
  )
};

export default ScrabbleTile;
