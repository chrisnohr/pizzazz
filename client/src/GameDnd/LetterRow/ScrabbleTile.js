import React, { PureComponent } from 'react';
import { LetterStyled, NumberSub } from './ScrabbleTileStyled';
import { letterValues } from '../../Constants';

class ScrabbleTile extends PureComponent {
  render() {
    const { letter, isValid } = this.props;

    return (
      <LetterStyled isValid={isValid} letter={letter}>
        { letter.toUpperCase() }
        <NumberSub>
          { letterValues[letter] }
        </NumberSub>
      </LetterStyled>
    );
  }
}

export default ScrabbleTile;
