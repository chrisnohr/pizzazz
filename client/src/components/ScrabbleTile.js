// @flow
import React, { PureComponent } from 'react';
import type { Node } from 'react';
import { LetterStyled, NumberSub } from './styled/ScrabbleTileStyled';
import { letterValues } from '../Constants';

type TileProps = {
  isValid: boolean,
  letter: string,
};

class ScrabbleTile extends PureComponent<TileProps> {
  render(): Node {
    const { letter, isValid } = this.props;

    return (
      <LetterStyled isValid={isValid} letter={letter}>
        { letter.toUpperCase() }
        <NumberSub>
          { letterValues[letter] }
        </NumberSub>
      </LetterStyled>
    );
  }
}

export default ScrabbleTile;
