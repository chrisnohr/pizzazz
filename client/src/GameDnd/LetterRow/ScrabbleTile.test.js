import React from 'react';
import { shallow } from 'enzyme';
import ScrabbleTile from './ScrabbleTile';
import { LetterStyled, NumberSub } from './ScrabbleTileStyled';
import { letterValues } from '../../Constants';

describe('<ScrabbleTile />', () => {
  it('renders the letter in uppercase', () => {
    const wrapper = shallow(<ScrabbleTile isValid={false} letter="e" />);
    expect(wrapper.find(LetterStyled).children().first().text()).toEqual('E');
  });

  it('renders the right letter value for z', () => {
    const wrapper = shallow(<ScrabbleTile isValid={false} letter="z" />);
    expect(wrapper.find(NumberSub).children().first().text()).toEqual(`${letterValues.z}`);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<ScrabbleTile isValid={false} letter="e" />);
    expect(wrapper).toMatchSnapshot();
  });
});
