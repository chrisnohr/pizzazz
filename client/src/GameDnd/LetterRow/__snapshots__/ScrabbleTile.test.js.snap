// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrabbleTile /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrabbleTile
    isValid={false}
    letter="e"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "E",
        <ForwardRef>
          1
        </ForwardRef>,
      ],
      "isValid": false,
      "letter": "e",
    },
    "ref": null,
    "rendered": Array [
      "E",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": 1,
        },
        "ref": null,
        "rendered": 1,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  line-height: normal;
  font-size: 2.6vw;
  font-weight: 400;
  vertical-align: -25%;
  @media screen and (min-width: 501px) {
    font-size:14.5px;
  }
",
            ],
          },
          "displayName": "styled.sub",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "sub",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  opacity: ",
          [Function],
          ";
  user-select: none;
  border: 1px solid ",
          "#CED4DA",
          ";
  border-radius: 0.5vw;
  color: ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  font-weight: 400;
  font-size: 8vw;
  text-align: center;
  line-height: 12.2vw;
  box-shadow: ",
          "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
          ";
  @media screen and (min-width: 501px) {
      font-size: 40px;
      line-height: 65px;
      border-radius: 4px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "E",
          <ForwardRef>
            1
          </ForwardRef>,
        ],
        "isValid": false,
        "letter": "e",
      },
      "ref": null,
      "rendered": Array [
        "E",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": 1,
          },
          "ref": null,
          "rendered": 1,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  line-height: normal;
  font-size: 2.6vw;
  font-weight: 400;
  vertical-align: -25%;
  @media screen and (min-width: 501px) {
    font-size:14.5px;
  }
",
              ],
            },
            "displayName": "styled.sub",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "sub",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  opacity: ",
            [Function],
            ";
  user-select: none;
  border: 1px solid ",
            "#CED4DA",
            ";
  border-radius: 0.5vw;
  color: ",
            [Function],
            ";
  background-color: ",
            [Function],
            ";
  font-weight: 400;
  font-size: 8vw;
  text-align: center;
  line-height: 12.2vw;
  box-shadow: ",
            "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
            ";
  @media screen and (min-width: 501px) {
      font-size: 40px;
      line-height: 65px;
      border-radius: 4px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
